instrumentation/kubernetes-client-7.0/src/main/java/io/opentelemetry/auto/instrumentation/kubernetesclient/KubernetesClientTracer.java

```diff
@@ -53,4 +52,33 @@ protected String requestHeader(Request request, String name) {
   protected String responseHeader(Response response, String name) {
     return response.header(name);
   }
+
+  @Override
+  protected Setter<Request> getSetter() {
+    return null;
+  }
+
+  @Override
+  protected String getInstrumentationName() {
+    return "io.opentelemetry.auto.kubernetes-client-7.0";
+  }
+
+  /** This method is overridden to allow other classes in this package to call it. */
+  @Override
+  protected Span onRequest(Span span, Request request) {
+    return super.onRequest(span, request);
+  }
+
+  /**
+   * This method is used to generate an acceptable CLIENT span (operation) name based on a given
+   * KubernetesRequestDigest.
+   */
+  public Span startSpan(KubernetesRequestDigest digest) {
```

Comment:
You don't need so much customisations here. You can use `startSpan(Request)` from the superclass. Override `spanNameForRequest` to get digest. You may need to override `startSpan(Request)` to add these two attributes from digest.

This also will remove the need to make `onRequest` public and to call it manually.

Reply:
this is old code.  If we want to refactor it, i can add it to the todo list.  It can be on a separate PR.